src/object/cli_ec.c

```diff
@@ -1254,11 +1254,16 @@ obj_ec_recx_reasb(daos_iod_t *iod, d_sg_list_t *sgl,
 		last = tgt_recx_idxs[i] + tgt_recx_nrs[i];
 	}
 	oiod->oiod_nr = idx;
-	*valid_tgt_nr = 0;
+	tmp_nr = update ? obj_ec_data_tgt_nr(oca) : tgt_nr;
 	for (i = 0, rec_nr = 0, last = 0; i < tgt_nr; i++) {
 		if (tgt_recx_nrs[i] == 0)
 			continue;
-		(*valid_tgt_nr)++;
+		if ((*single_data_tgt) && (i < tmp_nr)) {
+			if (*data_tgt_idx == -1)
+				*data_tgt_idx = i;
+			else if (*data_tgt_idx != i)
+				*single_data_tgt = false;
+		}
```

Comment:
I thought you can check tgt_bitmap to know if it write to single data tgt? then check the sgl or iod size to know whether it can do online ?  or I miss sth?

Reply:
right, using that bitmap also work, just need a few other code to check the bitmap (for example after your shard rotate patch need a few more code for this check), the code in this patch looks easier to understand?
but I am fine to change if you persist.

Reply:
oh, it is fine, since it is not easy to pass all tests.